dashboard/src/actions/authActions.js

```diff
@@ -7,6 +7,49 @@ import Cookies from "js-cookie";
 import { showToast } from "actions/toastActions";
 import { uid } from "../utils/helper";
 
+
+// Create an Authentication Request
+export const authenticationRequest = () => async (dispatch, getState) => {
+    try {
+      const endpoints = getState().apiEndpoint.endpoints;
+      const oidcServer = endpoints["openid-connect"]?.issuer;
+      const oidcRealm = endpoints["openid-connect"]?.realm;
+      const oidcClient = endpoints["openid-connect"]?.client;
+      // URI parameters ref: https://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint
+      // Refer Step 3 of pbench/docs/user_authentication/third_party_token_management.md
+      let req = oidcServer + '/realms/' + oidcRealm + '/protocol/openid-connect/auth';
+      req += '?client_id=' + oidcClient;
+      req += '&response_type=code';
+      req += '&redirect_uri=' + window.location.href.split('?')[0];
+      req += '&scope=profile';
+      req += '&prompt=login';
+      req += '&max_age=120';
+      window.location.href = req;
+    } catch (error) {
+      const alerts = getState().userAuth.alerts;
+      let alert = {};
+      if (error?.response) {
+        alert = {
+          title: error?.response?.data?.message,
+          key: uid(),
+        };
+        dispatch(toggleLoginBtn(true));
+      } else {
+        alert = {
+          title: error?.message,
+          key: uid(),
+        };
+        dispatch({ type: TYPES.OPENID_ERROR });
+      }
+      alerts.push(alert);
```

Comment:
I think that this `push()` is mutating the Redux state, which I believe we're supposed to avoid.  We're suppose to make a _copy_ of of the state that we intend to change, mutate it, and then hand the copy to the reducer via `dispatch()`.

Reply:
I am a little confused about what you mean by mutating the redux state, If this is an issue then we have to file a github/Jira issue to fix it in our code because there are a lot of places where we are mutating the `alerts` object without making the copy first.

Reply:
> there are a lot of places where we are mutating the alerts object

I only found two, and they are both in `dashboard//src/actions/authActions.js` (so, you could consider fixing them in this PR...they are both `push()` calls exactly like you're doing here).

> I am a little confused about what you mean by mutating the redux state

The stuff which comes back from `getState()` is supposed to be treated as read-only.  (Here's [one set of admonitions](https://redux.js.org/tutorials/fundamentals/part-4-store#inside-a-redux-store).)  It's only supposed to be changed in conjunction with doing a full state update (e.g., via a `dispatch()` to a "reducer").